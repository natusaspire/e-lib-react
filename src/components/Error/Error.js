import React, { memo } from 'react';

import Hero from '@/components/Hero';

const Error = ({ status }) => (
  <div className="main">
    <div className="main__hero">
      <Hero>
        <h1>Error {status}</h1>
      </Hero>
    </div>
  </div>
);

export default memo(Error);
