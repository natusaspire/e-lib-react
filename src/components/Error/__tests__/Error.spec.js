import React from 'react';
import { shallow } from 'enzyme';

import Error from '@/components/Error';

jest.mock('@/components/Hero', () => 'Hero');

describe('<Error />', () => {
  it('should render error info', () => {
    const wrapper = shallow(<Error status="404" />);

    expect(wrapper).toMatchSnapshot();
  });
});
