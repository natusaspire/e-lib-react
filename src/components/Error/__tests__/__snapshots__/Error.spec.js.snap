// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render correctly 1`] = `
<div
  className="main"
>
  <div
    className="main__hero"
  >
    <Hero>
      <h1>
        Error 
        404
      </h1>
    </Hero>
  </div>
</div>
`;
